build-logic-settings/build-environment/src/main/kotlin/gradlebuild/basics/BuildEnvironmentService.kt

```diff
@@ -58,9 +58,10 @@ abstract class BuildEnvironmentService : BuildService<BuildEnvironmentService.Pa
             DeprecationLogger.whileDisabled {
                 isIgnoreExitValue = true
             }
-            commandLine = listOf("git", *args)
             if (OperatingSystem.current().isWindows) {
-                commandLine = listOf("cmd.exe", "/d", "/c") + commandLine
+                commandLine(listOf("cmd.exe", "/d", "/c", "git", *args))
+            } else {
+                commandLine(listOf("git", *args))
```

Comment:
🤔 If we are thinking about these changes as examples people should follow, or at least dogfooding for ourselves to experience how changes will be made by users...

I think this would be better by turning `commandLine` into a local variable, and then assigning it to the property. That way the original code could stay mostly intact, resulting in a smaller change to history. WDYT?

Reply:
We can also do it that way. Done, check the new version